tempesta_fw/http_parser.c

```diff
@@ -1219,16 +1214,26 @@ __parse_connection(TfwHttpMsg *hm, unsigned char *data, size_t len)
 }
 
 /**
- * Parse Content-Length header value, RFC 2616 14.13.
+ * Parse Content-Length header value, RFC 7230 section 3.3.2.
  */
 static int
 __parse_content_length(TfwHttpMsg *msg, unsigned char *data, size_t len)
 {
 	int r;
 
 	/*
-	 * According to RFC 7230 section 3.3.2:
-	 *
+	 * A server MUST NOT send a Content-Length header field in any response
+	 * with a status code of 1xx (Informational) or 204 (No Content).  A
+	 * server MUST NOT send a Content-Length header field in any 2xx
+	 * (Successful) response to a CONNECT request
+	 */
+	if (TFW_CONN_TYPE(msg->conn) & Conn_Srv) {
+		TfwHttpResp *resp = (TfwHttpResp *)msg;
+
```

Comment:
Usually it's best to keep the code dense in cases like this, and don't add an extra empty line. It's just three lines of code. The first statement in the block is treated as an assignment in this case. Anyway, it's a matter of choice, preference, taste, etc.